department-of-veterans-affairs/caseflow
# Setup UAT data

Connects: https://github.com/department-of-veterans-affairs/caseflow/issues/4705 Lots of fun things in this PR.

1) Brining in VACOLS functions and procedures since we call at least one of them. The only things we don't have in our local vacols copy now are triggers and indexes.
2) Setting up a new environment called `local_vacols` to isolate this work until it's ready to be merged into our standard dev stream.
3) Scripts to dump data from UAT VACOLS, and seed them into the local database.
4) The beginning of an association between local_vacols data and VBMS data.
5) Timeshifting all date fields we pull in from a reference date of 2017-5-1, so the data looks current.
6) Hearing Prep, Reader, and Queue work with local_vacols

Note the data dumps will eventually be checked in, however, I need to work with Chris to ensure there is no PII leakage, until then you'll have to generate it yourself. See below.

**Test Plan**
1) If you haven't set it up already set up your local vacols using `rake local_vacols:setup`, otherwise just run `rake local_vacols:start` if it's been stopped.
1) In a shell run `ssh -L 1526:dsva-appeals-vacols-uat-markymark-2017-12-13-11-20.cdqbofmbcmtd.us-gov-west-1.rds.amazonaws.com:1526 <your ssh uat config>` which connects to my copy of VACOLS UAT for data dumping.
1) Run `RAILS_ENV=ssh_forwarding rake local_vacols:dump_data`
1) Run `RAILS_ENV=local_vacols rake local_vacols:seed`
1) Run `rails s -e local_vacols` to see the magic of our server using a local Oracle DB
1) Navigate to Reader, Hearing Prep, and Queue to ensure the data loads.
